monkey/infection_monkey/exploit/powershell.py

```diff
@@ -211,13 +210,18 @@ def _create_local_agent_file(self, binary_path):
             f.write(agent_binary_bytes.getvalue())
 
     def _run_monkey_executable_on_victim(self, executable_path):
+        set_agent_otp_command = (
+            f"set {AGENT_OTP_ENVIRONMENT_VARIABLE}={self.otp_provider.get_otp()}"
+        )
```

Comment:
For all exploiters, I would much prefer to put the OTP into the execute command itself (if possible). For example, on Linux, you can run with `OTP=boienstoaortsenarst /tmp/monkey-linux m0nk3y...`. On Windows I'm not sure of the exact syntax, but it should be roughly the same. This isolates the OTP environment variable to ONLY the agent process and its child processes.

Executing this is an atomic operation from the point of view of the agent. If we use the current approach, it's possible that we set the environment variable, then have some king of failure that prevents us from launching the agent. In this case, the environment variable won't be cleaned up (until the session successfully closes). This leaves the OTP open to being discovered by a malicious actor for longer.